monai/utils/nvtx.py

```diff
@@ -53,12 +56,46 @@ def __init__(
         name: Optional[str] = None,
         methods: Optional[Union[str, Tuple[str, ...]]] = None,
         append_method_name: Optional[bool] = None,
+        recursive: bool = False,
     ) -> None:
         self.name = name
         self.methods = methods
         self.append_method_name = append_method_name
+        self.recursive = recursive
 
     def __call__(self, obj: Any):
+        if self.recursive is True:
+            if isinstance(obj, Iterable):
+                annotated_transforms = [Range(recursive=True)(t) for t in obj]
+                return Range(self.name, recursive=False)(annotated_transforms)
+
+            from monai.transforms.compose import Compose, OneOf
+
+            if isinstance(obj, OneOf):
+                annotated_transforms = [Range(recursive=True)(t) for t in obj.transforms]
+                return Range(self.name, recursive=False)(
+                    OneOf(
+                        annotated_transforms,
+                        weights=obj.weights,
+                        map_items=obj.map_items,
+                        unpack_items=obj.unpack_items,
+                        log_stats=obj.log_stats,
+                    )
+                )
+
+            if isinstance(obj, Compose):
+                annotated_transforms = [Range(recursive=True)(t) for t in obj.transforms]
+                return Range(self.name, recursive=False)(
+                    Compose(
+                        annotated_transforms,
+                        map_items=obj.map_items,
+                        unpack_items=obj.unpack_items,
+                        log_stats=obj.log_stats,
+                    )
+                )
+
+            self.recursive = False
```

Comment:
could the overall recursion be simplified to this or similar ideas of in-place change, could you please help test out?

```suggestion
            if isinstance(obj, Iterable):
                return [Range(recursive=True)(t) for t in obj]

            from monai.transforms.compose import Compose, OneOf

            if isinstance(obj, Compose):
                obj.transforms = Range(recursive=True)(obj.transforms)
            self.recursive = False
```

Reply:
Yeah, that's cool. Except for a caveat that is missed in the unittests for Iterables. I'll update it.